src/code/z_lifemeter.c

```diff
@@ -442,18 +445,18 @@ void HealthMeter_Draw(GlobalContext* globalCtx) {
                 }
             }
 
-            temp3 = 26.0f + offsetY;
-            temp2 = 30.0f + offsetX;
-            temp4 = 1.0f;
-            temp4 /= 0.68f;
-            temp4 *= 1 << 10;
-            temp1 = 8.0f;
-            temp1 *= 0.68f;
-            gSPTextureRectangle(OVERLAY_DISP++, (s32)((temp2 - temp1) * 4), (s32)((temp3 - temp1) * 4),
-                                (s32)((temp2 + temp1) * 4), (s32)((temp3 + temp1) * 4), G_TX_RENDERTILE, 0, 0,
-                                (s32)temp4, (s32)temp4);
+            heartCenterY = 26.0f + offsetY;
+            heartCenterX = 30.0f + offsetX;
+            heartTexCoord = 1.0f;
+            heartTexCoord /= 0.68f;
+            heartTexCoord *= 1 << 10;
+            halfHeartLen = 8.0f;
+            halfHeartLen *= 0.68f;
+            gSPTextureRectangle(OVERLAY_DISP++, (s32)((heartCenterX - halfHeartLen) * 4), (s32)((heartCenterY - halfHeartLen) * 4),
+                                (s32)((heartCenterX + halfHeartLen) * 4), (s32)((heartCenterY + halfHeartLen) * 4), G_TX_RENDERTILE, 0, 0,
+                                (s32)heartTexCoord, (s32)heartTexCoord);
```

Comment:
```suggestion
                                (s32)heartTexCoordPerPixel, (s32)heartTexCoordPerPixel);
```

A botched job of an explanation:

[`gSPTextureRectangle`](http://n64devkit.square7.ch/n64man/gsp/gSPTextureRectangle.htm)

`1 / 0.68 * (1 << 10)` represents `1 / 0.68` in the fixed-point s5.10 format
that means every screen pixel, the texture coordinate advances by `1 / 0.68 = 1.47` image pixel

so by drawing over a `halfHeartLen*2`-wide screen pixels area the texture coordinate changes by `halfHeartLen*2 * 1/0.68`
(with `halfHeartLen = 8 * 0.68`) `halfHeartLen*2 * 1/0.68 = 16` image pixels, which is indeed the width of eg `gHeartFullTex` (as can be seen in `assets/xml/textures/parameter_static.xml`)